smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/model/SymbolExt.kt

```diff
@@ -54,6 +57,31 @@ val Symbol.isBoxed: Boolean
 val Symbol.isNotBoxed: Boolean
     get() = !isBoxed
 
+enum class PropertyTypeMutability {
+    /**
+     * Immutable property (e.g. `val`)
+     */
+    IMMUTABLE,
+
+    /**
+     * Mutable property (e.g. `var`)
+     */
+    MUTABLE;
+
+    override fun toString(): String = when (this) {
+        IMMUTABLE -> "val"
+        MUTABLE -> "var"
+    }
+}
+
+/**
+ * Get the property type mutability of this symbol if set.
+ */
+val Symbol.propertyTypeMutability: PropertyTypeMutability?
+    get() = getProperty(SymbolProperty.PROPERTY_TYPE_MUTABILITY)
+        .map { it as PropertyTypeMutability }
+        .getOrNull()
```

Comment:
I understand the nuance here you are getting at but most documentation out there refers to `val` vs `var` as immutable vs mutable. Which is still accurate w.r.t assignment of a variable (a `val` variable can only be assigned once and is then considered "immutable" (w.r.t assignment)). That of course doesn't mean the type assigned can't be mutated but it is immutable w.r.t assignment. I'm inclined to stick with this verbiage since it is fairly pervasive out there. 

e.g. Official kotlin [docs](https://kotlinlang.org/docs/properties.html#declaring-properties) referring to `var` as mutable.

Reply:
That's disappointing/inconsistent since [elsewhere](https://kotlinlang.org/docs/basic-syntax.html#variables) their docs use words like "read-only" and "can be reassigned". In my mind, mutability and immutability strictly refer to internal state not external references but that appears to not be universal.

Point withdrawn. 😊